src/core/wallet/types.ts

```diff
@@ -14,6 +14,7 @@ export enum TransactionStatus {
 }
 
 export interface IWallet {
-    getAccounts(blockchain: Blockchain, index: number, indexTo: number): Promise<IAccountState[]>;
+    getAccounts(blockchain: Blockchain, index: number, indexTo?: number): Promise<IAccountState[]>;
     sign(blockchain: Blockchain, accountIndex: number, tx: IBlockchainTransaction): Promise<string>;
+    onAppOpened?(blockchain: Blockchain);
```

Comment:
i wouldn't add this on the interface. `onAppOpened` is specific to hw wallets, so i would remove this and where you use HWWallet instance and need to call this method just cast to HWWallet